qemu/tests/qemu_guest_agent.py

```diff
@@ -1849,6 +1853,124 @@ def gagent_check_with_migrate(self, test, params, env):
+
+            :param cmd: cmd to be wrapped.
+            :return: wrapped cmd
+            """
+            disk = "disk_" + ''.join(random.sample(string.ascii_letters +
+                                                   string.digits, 4))
+            cmd_header = "echo list disk > " + disk
+            cmd_header += " && echo select disk %s >> " + disk
+            cmd_footer = " echo exit >> " + disk
+            cmd_footer += " && diskpart /s " + disk
+            cmd_footer += " && del /f " + disk
+            cmd_list = []
+            for i in cmd.split(";"):
+                i += " >> " + disk
+                cmd_list.append(i)
+            cmd = " && ".join(cmd_list)
+            return " && ".join([cmd_header, cmd, cmd_footer])
+
+        session = self._get_session(params, self.vm)
+        image_size_stg0 = params["image_size_stg0"]
+        is_rhel6 = params.get("is_rhel6")
+
+        error_context.context("Format the new data disk and mount it.",
+                              logging.info)
+        if params.get("os_type") == "linux":
+            disk_data = list(utils_disk.get_linux_disks(session).keys())
+            mnt_point = utils_disk.configure_empty_disk(
+                session, disk_data[0], image_size_stg0, "linux",
+                labeltype="msdos")
+            partition = "/dev/%s" % mnt_point[0].split("/")[-1]
+        else:
+            disk_index = utils_misc.wait_for(
+                lambda: utils_disk.get_windows_disks_index(session,
+                                                           image_size_stg0),
+                120)
+            if disk_index:
+                logging.info("Clear readonly for disk and online it in"
+                             " windows guest.")
+                if not utils_disk.update_windows_disk_attributes(session,
+                                                                 disk_index):
+                    test.error("Failed to update windows disk attributes.")
+                mnt_point = utils_disk.configure_empty_disk(
+                    session, disk_index[0], image_size_stg0, "windows",
+                    labeltype="msdos")
+
+        error_context.context("Freeze fs.", logging.info)
+        self.gagent.fsfreeze()
+
+        error_context.context("Umount fs or offline disk in guest after"
+                              " fsfreeze.", logging.info)
+        if params.get("os_type") == "linux":
+            # for rhel6 guest, umount will fail as timeout.
+            try:
+                umount_cmd = "umount %s" % mnt_point[0]
+                session.cmd_status(umount_cmd)
+            except ShellTimeoutError:
+                if is_rhel6 == "yes":
+                    logging.info("Umount cmd timeout for rhel6 guest as"
+                                 " expected.")
+                else:
+                    test.fail("Umount cmd timeout for rhel7+ guest.")
+            else:
+                test.fail("Umount fs failed after fsfreeze.")
```

Comment:
Umount fs `should` failed after fsfreeze?

Reply:
For rhel6 guest,it will fail.